### PR TITLE
Fixes #8151 Build flavors for Amazon Appstore and Google Playstore

### LINKED ISSUE #8151: Amazon AppStore publish rejected for camera permission request
###### Reproduction Steps

1. Attempt to submit a build of AnkiDroid from release-2.14 branch to amazon app store
2. 
3. 

###### Expected Result

Submission successful!

###### Actual Result

```
Your app submission does not meet one or more of our acceptance criteria for some or all targeted devices. Failure reason(s) are listed below:

    To ensure the best user experience apps may not request permissions for capabilities not present within Amazon Fire tablet or Fire TV devices. Camera functionality is not currently supported on Fire TV devices. Therefore, please remove the camera permission request and resubmit your app. For additional details see the Test Criteria for Amazon Appstore Apps section of the Appstore Developer Console.

You can view our guidelines on the Appstore Developer Portal. 
```

https://developer.amazon.com/docs/app-submission/faq-submission.html#device-targeting

:shrug: 

###### Debug info
Refer to the [support page](https://ankidroid.org/docs/help.html) if you are unsure where to get the "debug info".

###### Research
*Enter an [x] character to confirm the points below:*

- [ ] I have read the [support page](https://ankidroid.org/docs/help.html) and am reporting a bug or enhancement request specific to AnkiDroid
- [ ] I have checked the [manual](https://ankidroid.org/docs/manual.html) and the [FAQ](https://github.com/ankidroid/Anki-Android/wiki/FAQ) and could not find a solution to my issue
- [ ] I have searched for similar existing issues here and on the user forum
- [ ] (Optional) I have confirmed the issue is not resolved in the latest alpha release ([instructions](https://docs.ankidroid.org/manual.html#betaTesting))

### PR DESCRIPTION
**Pull request by mrudultora**
## Purpose / Description
Different flavors added for Amazon App store and Google Play store (we cannot use default here). And, I tried to hide the camera related options in amazon flavor.

## Fixes
Fixes #8151

## Approach
Added the flavors in gradle. Created a class to get all the permissions to find out whether camera permission is present in manifest or not. If not present. hide the camera related options from menu's.
4 Build variants are generated now (amazonDebug, amazonRelease, playDebug, playRelease)

img

## How Has This Been Tested?
Tested on real device (Redmi Note 7S and API 29).

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code
- [x] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [x] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by mrudultora** [comment]
Apologies, as I messed up with rebase on #8302 I have created this PR with all the required changes.

> As Mike suggested, I added Camera Permission in our main manifest. And also removed the AndroidManifest for play flavor. Renaming play flavor to default is not possible as this keyword is not allowed in gradle for naming purposes. Also changed the method name. It's working as intended. Tested both the flavors on real device.

File: AnkiDroid/src/main/java/com/ichi2/utils/CheckCameraPermission.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+            String[] requestedPermissions = context.getPackageManager()
+                    .getPackageInfo(context.getPackageName(), PackageManager.GET_PERMISSIONS)
+                    .requestedPermissions;
+            if (Arrays.toString(requestedPermissions).contains("android.permission.CAMERA")) {
+                return false;
+            }
+        } catch (Exception e) {
+            Timber.w(e);
+        }
+        return true;
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/utils/CheckCameraPermission.java:40]:
```suggestion
    public static boolean manifestContainsPermission(Context context) {
        try {
            String[] requestedPermissions = context.getPackageManager()
                    .getPackageInfo(context.getPackageName(), PackageManager.GET_PERMISSIONS)
                    .requestedPermissions;
            if (Arrays.toString(requestedPermissions).contains("android.permission.CAMERA")) {
                return true;
            }
        } catch (Exception e) {
            Timber.w(e);
        }
        return false;
```

File: AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/MultimediaEditFieldActivity.java [hunk 2]
```diff
@@ -215,6 +216,11 @@ public boolean onCreateOptionsMenu(Menu menu) {
         menu.findItem(R.id.multimedia_edit_field_to_audio).setVisible(mField.getType() != EFieldType.AUDIO_RECORDING);
         menu.findItem(R.id.multimedia_edit_field_to_audio_clip).setVisible(mField.getType() != EFieldType.AUDIO_CLIP);
         menu.findItem(R.id.multimedia_edit_field_to_image).setVisible(mField.getType() != EFieldType.IMAGE);
+
+        /* To check whether Camera Permission is asked in AndroidManifest.xml */
+        if (CheckCameraPermission.isPermissionAvailableInManifest(this)) {
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/multimediacard/activity/MultimediaEditFieldActivity.java:221]:
```suggestion
        if (!CheckCameraPermission.manifestContainsPermission(this)) {
```

File: AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java [hunk 3]
```diff
@@ -1681,6 +1682,13 @@ private void setMMButtonListener(ImageButton mediaButton, final int index) {
                 PopupMenuWithIcons popup = new PopupMenuWithIcons(NoteEditor.this, v, true);
                 MenuInflater inflater = popup.getMenuInflater();
                 inflater.inflate(R.menu.popupmenu_multimedia_options, popup.getMenu());
+
+                /* To check whether Camera Permission is asked in AndroidManifest.xml */
+                if (CheckCameraPermission.isPermissionAvailableInManifest(this)) {
```
Comment (mikehardy) [inline AnkiDroid/src/main/java/com/ichi2/anki/NoteEditor.java:1687]:
```suggestion
                if (!CheckCameraPermission.manifestContainsPermission(this)) {
```

**Review by mikehardy — APPROVED** [review]
Certainly seems like it should work - whatever we do with Amazon it will rely on a separate flavor, so this machinery will be required either way - this was a big help @mrudultora thank you!

**Pull request merged**